### ISSUE TITLE
 Git for Windows 2.26.1 and up crash on "git log" custom date format (--pretty='format:%cd')

### ISSUE THREAD

**Issue by 3c0** [git-for-windows/git#3624]
### Setup

```
$ git --version --build-options

git version 2.34.1.windows.1
cpu: i686
built from commit: 2ca94ab318509b3c271e82889938816bad76dfea
sizeof-long: 4
sizeof-size_t: 4
shell-path: /bin/sh
feature: fsmonitor--daemon
```

 - Which version of Windows are you running? Vista, 7, 8, 10? Is it 32-bit or 64-bit?

```
$ cmd.exe /c ver

Microsoft Windows [version 10.0.19044.1415]
```

 - What options did you set as part of the installation? Or did you choose the
   defaults?

```
# One of the following:
> type "C:\Program Files\Git\etc\install-options.txt"

Editor Option: VIM
Custom Editor Path:
Default Branch Option:
Path Option: Cmd
SSH Option: OpenSSH
Tortoise Option: false
CURL Option: OpenSSL
CRLF Option: CRLFAlways
Bash Terminal Option: MinTTY
Git Pull Behavior Option: Rebase
Use Credential Manager: Disabled
Performance Tweaks FSCache: Enabled
Enable Symlinks: Enabled
Enable Pseudo Console Support: Disabled
Enable FSMonitor: Disabled
```

### Details

 - Which terminal/shell are you running Git from? e.g Bash/CMD/PowerShell/other

GNU bash, version 4.4.23(1)-release (i686-pc-msys)

 - What commands did you run to trigger this issue? If you can provide a
   [Minimal, Complete, and Verifiable example](http://stackoverflow.com/help/mcve)
   this will help us understand the issue.

```
git log --date='format:%Y-%m-%d %H:%M' --pretty='format:%cd'
```
 - What did you expect to occur after running these commands?

Listing of commit dates.

 - What actually happened instead?

Segmentation fault

- Origin of bug :

[Git for Windows 2.26.1](https://github.com/git-for-windows/git/releases/tag/v2.26.1.windows.1)
**Bug Fixes** :  Git now [accepts more date formats](https://github.com/git-for-windows/git/pull/2574) such as %g and %V.

- Last version not crashing is 2.26.0.

- Maybe related: #2495 fix (https://github.com/git-for-windows/git/commit/8e892d0b86f4c7c280ca7fef3ab95d906750cc40)

**Comment by dscho** [comment]
> `git log --date='format:%Y-%m-%d %H:%M' --pretty='format:%cd'`

Note that this works in x86_64 builds, but crashes in i686 builds.

**Comment by dscho** [comment]
I suspect that this might be fixed by https://github.com/git/git/pull/1181, and am testing that theory now.

**Comment by dscho** [comment]
> I suspect that this might be fixed by [git#1181](https://github.com/git/git/pull/1181), and am testing that theory now.

Turns out that is exactly the case. Git for Windows PR is up: https://github.com/git-for-windows/git/pull/3626

**Comment by 3c0** [comment]
> Turns out that is exactly the case. Git for Windows PR is up: #3626

Wow! Perfect timing and thank you @dscho for the prompt inquiry and resolution.
